clients/src/main/java/org/apache/kafka/common/requests/ListOffsetsRequest.java

```diff
@@ -82,8 +82,13 @@ else if (requireTimestamp)
             return new Builder(minVersion, ApiKeys.LIST_OFFSETS.latestVersion(), CONSUMER_REPLICA_ID, isolationLevel);
         }
 
+        public static Builder forReplica(short allowedVersion, int replicaId, boolean requireEarliestLocalTimestamp) {
+            short minVersion = requireEarliestLocalTimestamp ? (short) 8 : (short) 0;
```

Comment:
@FrankYang0529 : Yes, the receiver does have the check. I am just saying there is no need to add the code here since it's intended for the sender, but the sender doesn't execute the added logic.